src/kokkos/plugin-PixelTriplets/kokkos/CAHitNtupletGeneratorKernels.cc

```diff
@@ -167,20 +167,20 @@ namespace KOKKOS_NAMESPACE {
     // in principle we can use "nhits" to heuristically dimension the workspace...
     device_isOuterHitOfCell_ =
         Kokkos::View<GPUCACell::OuterHitOfCell *, KokkosExecSpace>("device_isOuterHitOfCell_", std::max(1U, nhits));
-#ifdef TODO
-    {
-      int threadsPerBlock = 128;
-      // at least one block!
-      int blocks = (std::max(1U, nhits) + threadsPerBlock - 1) / threadsPerBlock;
-      gpuPixelDoublets::initDoublets<<<blocks, threadsPerBlock, 0, stream>>>(device_isOuterHitOfCell_.get(),
-                                                                             nhits,
-                                                                             device_theCellNeighbors_,
-                                                                             device_theCellNeighborsContainer_.get(),
-                                                                             device_theCellTracks_,
-                                                                             device_theCellTracksContainer_.get());
-      cudaCheck(cudaGetLastError());
-    }
+
+#ifdef KOKKOS_BACKEND_CUDA
+    int teamSize = 128;
+    // at least one league
+    int leagueSize = (std::max(1U, nhits) + teamSize - 1) / teamSize;
+    Kokkos::TeamPolicy<KokkosExecSpace> policy{execSpace, leagueSize, teamSize};
+#else
+    Kokkos::TeamPolicy<KokkosExecSpace> policy{execSpace, 1, Kokkos::AUTO()};
```

Comment:
I think it would be better to do have these the other way around, if `KOKKOS_BACKEND_SERIAL`, then use `1, Kokkos::AUTO()`, and else use many teams. Then the "many teams" would get applied automatically for all parallel backends.

Reply:
sure, this kernel can be easily reorganized in that way

Reply:
Actually a question here: how can we properly set `N` in `TeamPolicy(N, AUTO)` for different backends? I thought Kokkos may provide an auto-tuning abstraction for the whole team policy (i.e.,  `TeamPolicy(AUTO)`) but looks like `N` has to be set depending on each specific backend. In that case, a generic parallel solution seems impossible and we still need `#ifdef ...` to specify `N` manually.

Reply:
I think this is a good question for the notes :) You're right (in the other comment) that for immediate purposes Serial and CUDA backends are sufficient, but we should also keep in mind the other backends. For practical purposes this means just taking `N` as the number of blocks in CUDA for others-than-serial, and `1` for serial.

I'm probably biased from the flexibility in TBB and CUDA for not even thinking why `N` teams would not work more than `N` worker threads (in CPU case).

Reply:
Thanks, I will discuss this in the notes.